skll/logutils.py

```diff
@@ -53,5 +58,38 @@ def get_skll_logger(name, filepath=None, log_level=logging.INFO):
             file_handler.setLevel(log_level)
             logger.addHandler(file_handler)
 
+    # Make method for sending sklearn-specific warnings to the
+    # logger that can be used to replace warnings.showwarning
+    def send_sklearn_warnings_to_log(message, category, filename, lineno,
```

Comment:
There doesn't seem to be a real reason why this function is defined _inside_ `get_skll_logger()` other than just the fact that we want the function in scope? Perhaps a better solution would be as follows:

1. Rename `send_sklearn_warnings_to_log()` to `send_sklearn_warnings_to_logger()`, make it a top-level function in `logutils.py`, and add a first argument `logger`. 
2. Use the following assignment statement in `get_skll_logger()`: 

    ```python
    warnings.showwarning = partial(send_sklearn_warnings_to_log, logger)
    ```

This would make it so that there's a clear connection between the logger to be returned and the logger to which scikit-learn warnings should be sent. 

I just tested this and it seems to work. Thoughts @mulhod ?

Reply:
Seems to be working!